packages/react-components/src/components/CheckBox.tsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { useRuleEngine, useFormIntl } from '@aemforms/af-react-renderer';
+import {State, FieldJson} from '@aemforms/af-core';
+import '../../dist/theme.css';
+
+const CheckBox = (fieldjson: State<FieldJson>) => {
+   const [props, handlers] = useRuleEngine(fieldjson); 
+   const { id, label, enum: enums,  value, valid, errorMessage, description, visible, name, readOnly } = props;  
+   const i18n = useFormIntl();
+   const selectedValue = enums?.[0];
+   const unSelectedValue = (enums?.length || 0) < 2 ? null : enums?.[1];
+   const validateState = valid === false ? 'invalid' : 'valid';
+   const error = validateState === 'invalid';
+   const formattedDescription = i18n.formatMessage({ id: `${description}` });
+   const formattedLabel = i18n.formatMessage({ id: `${label?.value}` });
+   
+   const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const val = e.target.checked ? selectedValue : unSelectedValue;
+      handlers.dispatchChange(val);
+   };
+   return visible ? (
+     <div>
+       <input type='checkbox' onChange={handleChange} value={value} name={name} readOnly={readOnly} />
+       <span>{formattedLabel}</span>
```

Comment:
Try to use a label tag instead of span
incorporate "label.visible" property

Reply:
Implemented!